### PR TITLE
Update harden_server.rst - improve fail2ban section

### LINKED ISSUE #12327: Fail2ban instructions in hardening guide miss needed loglevel and some details
Nextcloud won't log failed login attempts in log levels > ``2``.
I added this to the documentation, and, while at it, improved the fail2ban guide in a few other places.

### PR DESCRIPTION
**Pull request by tgoeg**
Nextcloud won't log failed login attempts in log levels > 2. Added this to the documentation, and, while at it, improved the fail2ban guide in a few other places.

See https://github.com/nextcloud/server/issues/48826 as well.

### ☑️ Resolves

* Fix #12327

### 🖼️ Screenshots
2024-10-28_131057_screenshot

### CODE REVIEW

**Review by susnux — APPROVED** [review]
Makes sense

File: admin_manual/installation/harden_server.rst [hunk 1]
```diff
@@ -297,6 +297,26 @@ Setup fail2ban
...
+system level, so the webserver, PHP and the database do not need to handle this
+unnecessary traffic at all.
+
+Nextcloud prerequisites
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Nextcloud logs failed login attempts in ``nextcloud.log`` with log level ``2``,
+so you need to define a ``loglevel`` of ``2`` or less in ``config.php``.
+
+Make sure your ``nextcloud.log`` is writeable by your webserver user, possibly by
```
Comment (nickvergessen) [inline admin_manual/installation/harden_server.rst:310]: In theory the nextcloud.log is owned by the webserver user anyway, as it's the one creating and writing it?

Comment (tgoeg) [reply admin_manual/installation/harden_server.rst:310]: Only in theory :-)
Mine is not, as the whole installation is owned by a specific linux user as I never let the `www-data` user own its own executables (r/w), as that opens up the possibility for exploits to change the application's code.
Most admins I know do it this way.

**Pull request merged**